src/app/forget-password/page.tsx

```diff
@@ -0,0 +1,101 @@
+'use client'
+import { signIn, useSession } from 'next-auth/react'
+import { useRouter } from 'next/navigation'
+import React, { useState, useEffect, FormEvent } from 'react'
+import Link from 'next/link'
+
+const ResetPassword = () => {
+  const router = useRouter()
+  const [error, setError] = useState('')
+  const { data: session, status: sessionStatus } = useSession()
+
+  useEffect(
+    () => {
+      if (sessionStatus === 'authenticated') {
+        router.replace('/dashboard')
+      }
+    },
+    [sessionStatus],
+    router,
+  )
+
+  const isValidEmail = (email: string) => {
+    const emailRegex = /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i
+    return emailRegex.test(email)
+  }
+
+  const handleSubmit = async (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault()
+    const email = e.currentTarget.email.value
+
+    if (!isValidEmail(email)) {
+      setError('Email is invalid')
+      return
+    }
+    try {
+      const res = await fetch('api/auth/forget-password', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          email,
+        }),
+      })
+
+      if (res.status === 400) {
+        setError('User with this email is not registered.')
+      }
+      if (res.status === 200) {
+        setError('')
+        router.push('/')
+      }
+    } catch (error) {
+      setError('Error, try again')
+      console.log(error)
+    }
+  }
+
+  if (sessionStatus === 'loading') {
+    return <h1>Loading...</h1>
+  }
```

Comment:
handling password reset requests. Validates email, sends a reset request to the server, and redirects to the login page.